### ISSUE TITLE
what's different between forwarder and modifier?

### ISSUE THREAD

**Issue by chennqqi** [s8sg/faas-flow#48]
a modifier:   NODE[ function ...] -> (async to) -> NODE [ modifier ] -> NODE...
a forwarder:   NODE[function -> forwarder]  -> (async to ) -> NODE...

**Comment by s8sg** [comment]
A modifier is an `Operation` just like a `function` or `callback`. 
Its basically a `inline function` in flow definition 
Now operation can be executed in a node, once or multiple times (by chaining). 
So `Node A` can have multiple `Operations` and `Modifier` can be one or many of such `Operations`
```yaml
Node A:
   Operations:
       -  Modifier
       -  Modifier
       -  Function
       -  Callback
       -  Modifier
       -   ....
```

`Forwader` is very different concept. Its allows to modify requests between different nodes. 
For example, after execution of `Node A` it data request to `NodeB` and `NodeC`. Now `Forwarder ` allows to modify the data for different `Nodes`, so that `Node B` and `NodeC` can possibly get two different data.
```yaml
Node A:
   Children:
      NodeB:
         - Forwarder
      NodeC:
         - Forwarder
``` 

In Summary:
A modifier is an `Operation` defined as a `inline function` in a Node
A forwarder is a data Handler for each of the different `Edges` form a Node

**Comment by chennqqi** [comment]
![image](https://user-images.githubusercontent.com/5073720/59992524-d5e34d80-967e-11e9-8e57-3999319b0d6b.png)
Is My understanding right?

**Comment by s8sg** [comment]
Right. 

We have a map of Forwarder for each nodes as
```go
// maps ChildrenId -> Forwarder
map[string]Forwarder
```
Currently for each Children by default we provide a `Forwarder` as:
```go
fun DefaultForwader(data []byte) []byte { return data }
```
To mark that data doesn't get forwarded for a edge, we simply set the Forwarder as `nil` for that childId.
To make it easy for User we added a simpler Option
```go
faasflow.Execution
```
